canonical-web-and-design/maas-ui
# fix(legacy): add type=button to cancel buttons

## Done

Without type="button" set, many cancel buttons are the default action when pressing 'enter' in the app, instead of submitting or saving the form.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit `/MAAS/l/networks?by=fabric`
- Select 'add fabric' from the dropdown
- Enter some text for a new fabric
- Hit enter and ensure the form as submitted, rather then cancelling

## Fixes

Fixes: #1809

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
